django/test/testcases.py

```diff
@@ -379,14 +382,15 @@ def _setup_and_call(self, result, debug=False):
                 result.addError(self, sys.exc_info())
                 return
 
-    def _pre_setup(self):
+    @classmethod
```

Comment:
I'll rebase this change into the first commit, so that the first commit has all three methods that became either class or static.